src/components/menu/menu-item/menu-item.component.js

```diff
@@ -1,33 +1,152 @@
-import React from 'react';
+import React, {
+  useRef, useState, useEffect
+} from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import StyledMenuItemWrapper from './menu-item.style';
 import { StyledSubmenu, StyledSubmenuItem, StyledSubmenuTitle } from '../submenu-block/submenu.style';
 import OptionHelper from '../../../utils/helpers/options-helper';
 import Link from '../../link';
+import Events from '../../../utils/helpers/events';
 
-const MenuItem = ({
+const MenuItem = React.forwardRef(({
   submenu,
   children,
   href,
   to,
   menuType,
   onClick,
   target,
-  submenuDirection,
+  submenuDirection = 'right',
   icon,
   selected,
-  routerLink
-}) => {
+  routerLink,
+  isFirstElement,
+  handleKeyDown
+}, ref) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [openByArrowUp, setOpenByArrowUp] = useState(false);
+  const submenuItemsRefs = submenu && useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  const ifRef = () => !submenu && { ref };
+  let actualIndex;
+  const setFocusToSubmenuElement = (event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    const findIndex = () => {
+      if (submenuItemsRefs.current[index] && submenuItemsRefs.current[index].current.tagName === 'DIV') {
+        if (Events.isDownKey(event)) {
+          return index + 1;
+        }
+        return index - 1;
+      }
+      return index;
+    };
+    const tempIndex = findIndex();
+    let newIndex = tempIndex;
+    actualIndex = tempIndex;
+
+    if (index === submenuItemsRefs.current.length) {
```

Comment:
it would be great to define a variable that describes the condition e.g.

```js
const isFocusOnLastMenuItem = index === submenuItemsRefs.current.length;
if(isFocusOnLastMenuItem){...}
```